unified-runtime/scripts/core/exp-command-buffer.yml

```diff
@@ -1166,39 +1169,46 @@ returns:
 --- #--------------------------------------------------------------------------
 type: function
 desc: "Update a kernel launch command in a finalized command-buffer."
-details: "This entry-point is synchronous and may block if the command-buffer is executing when the entry-point is called."
+details: "This entry-point is synchronous and may block if the command-buffer is executing when the entry-point is called. On error, the state of the command-buffer commands being updated is undefined."
 class: $xCommandBuffer
 name: UpdateKernelLaunchExp
```

Comment:
Since we're breaking this API, i'm thinking we future proof it by renaming it `UpdateMutableCommandsExp` and copy the design from [clUpdateMutableCommandsKHR](https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_API.html#clUpdateMutableCommandsKHR) so that different command types other than kernels could be passed for update in future (although we'd only still support kernels in the scope of this PR)

Reply:
The idea is that you can pass as two separate parameters, list of `void*` ptrs and a list of enums identifying the types to cast those pointers to at the same index in the list. So our current struct will be the only enum value and struct type defined for casting, but in the future we can add new enums and types for casting entries to.

Reply:
Makes sense to me!